### PR TITLE
Use pyproject.toml instead of setup.py

### PR DESCRIPTION
**Pull request by jl-wynen**
Tested in https://github.com/scipp/scipp/actions/runs/6508607013
I downloaded the build wheel and it works on my machine.

~Needs some cleanup...~

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,13 +1,84 @@
 [build-system]
 requires = [
-  "setuptools>=42",
-  "wheel",
-  "cmake>=3.21",
-  "conan==1.57.0",
-  "ninja",
-  "scikit-build==0.13.1",
+  "setuptools>=68",
+  "conan==1.61.0",
```
Comment (jl-wynen) [inline pyproject.toml:4]: This version differs from our conda setup. I needed to get a newer version here to make it work with scikit build. But conda still only has 1.59.

We need to update to version 2...

File: setup.py [hunk 2]
```diff
@@ -1,108 +0,0 @@
...
-from setuptools import find_packages
-from skbuild import setup
-
-
-def get_version():
-    import subprocess
-
-    return (
-        subprocess.run(
-            ['git', 'describe', '--tags', '--abbrev=0'], stdout=subprocess.PIPE
```
Comment (SimonHeybrock) [inline setup.py:13]: Remind me how this gets the version now? Was it that code snipped we have in our `__init__.py` files?

Comment (jl-wynen) [reply setup.py:13]: The same as all other packages. It uses `setuptools_scm` which essentially runs the code here. Differently from the other projects, here, we write the version to `src/scipp/_version.py` to allow scikit-build to pick it up. (I don't know how this works in pure setuptools-based builds.) However, the actual `scipp.__version__` is based on C++. There is some code in CMakeLists.txt - that I didn't touch - that also runs this git command and makes it so the version gets written into the binary. So the change here is only for the version reported in PyPI.

Comment (SimonHeybrock) [reply setup.py:13]: > It uses setuptools_scm which essentially runs the code here.

Which code here? The file was removed.

Comment (jl-wynen) [reply setup.py:13]: This code you highlighted to query git.

Comment (SimonHeybrock) [reply setup.py:13]: So by "runs the code here", you do not mean the *code here* (which was deleted) but *runs the same command*?

Comment (jl-wynen) [reply setup.py:13]: Yes, or a similar one. It uses git to get the latest tag and extract a version from it. (We've been using it in our python-only projects. But without writing a `_version.py` file.)

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**